ui-ngx/src/app/modules/home/components/filter/key-filter-dialog.component.ts

```diff
@@ -121,22 +121,30 @@ export class KeyFilterDialogComponent extends
       this.keyFilterFormGroup.get('valueType').valueChanges.pipe(
         takeUntil(this.destroy$)
       ).subscribe((valueType: EntityKeyValueType) => {
-        const prevValue: EntityKeyValueType = this.keyFilterFormGroup.value.valueType;
+        const prevValueType: EntityKeyValueType = this.keyFilterFormGroup.value.valueType;
         const predicates: KeyFilterPredicate[] = this.keyFilterFormGroup.get('predicates').value;
-        if (prevValue && prevValue !== valueType && predicates && predicates.length) {
-          this.dialogs.confirm(this.translate.instant('filter.key-value-type-change-title'),
-            this.translate.instant('filter.key-value-type-change-message')).subscribe(
-            (result) => {
-              if (result) {
-                this.keyFilterFormGroup.get('predicates').setValue([]);
-              } else {
-                this.keyFilterFormGroup.get('valueType').setValue(prevValue, {emitEvent: false});
+        const value = this.keyFilterFormGroup.get('value')?.value;
+        if (prevValueType && prevValueType !== valueType) {
+          if (this.isConstantKeyType && this.data.telemetryKeysOnly) {
+            this.keyFilterFormGroup.get('value').setValue(null);
+          }
+          if (predicates && predicates.length) {
+            this.dialogs.confirm(this.translate.instant('filter.key-value-type-change-title'),
+              this.translate.instant('filter.key-value-type-change-message')).subscribe(
+              (result) => {
+                if (result) {
+                  this.keyFilterFormGroup.get('predicates').setValue([]);
+                } else {
+                  this.keyFilterFormGroup.get('valueType').setValue(prevValueType, {emitEvent: false});
+                  this.keyFilterFormGroup.get('value')?.setValue(value, {emitEvent: false});
+                }
               }
-            }
-          );
+            );
+          }
         }
-        if (valueType === EntityKeyValueType.BOOLEAN && this.isConstantKeyType) {
+        if (valueType === EntityKeyValueType.BOOLEAN && this.isConstantKeyType && this.data.telemetryKeysOnly) {
```

Comment:
It is necessary to change the order of predicates in the condition
```suggestion
        if (this.data.telemetryKeysOnly && this.isConstantKeyType && valueType === EntityKeyValueType.BOOLEAN) {
```